be/src/storage/persistent_index.cpp

```diff
@@ -1800,8 +1800,13 @@ Status ShardByLengthMutableIndex::load(const MutableIndexMetaPB& meta) {
     for (int i = 0; i < n; i++) {
         const auto& page_pointer_pb = meta.wals(i).data();
         auto offset = page_pointer_pb.offset();
-        for (const auto& shard : _shards) {
-            RETURN_IF_ERROR(shard->load(offset, read_file));
+        std::string buff;
+        raw::stl_string_resize_uninitialized(&buff, 4);
+        RETURN_IF_ERROR(read_file->read_at_fully(offset, buff.data(), buff.size()));
+        const auto key_size = UNALIGNED_LOAD32(buff.data());
+        const auto [shard_offset, shard_size] = _shard_info_by_key_size[key_size];
+        for (auto i = 0; i < shard_size; ++i) {
+            RETURN_IF_ERROR(_shards[shard_offset + i]->load(offset, read_file));
```

Comment:
we separate kvs to different length in one WAL, so we need to load kvs of different length but not the first length. I think these codes should be in a loop.